### PR TITLE
Ap create new reminder

### LINKED ISSUE #1: See All Reminders
When the user visits the root of the application (e.g. the "/" URL), they should see a list of all of the reminders on the page.

- Each reminder should have a class of .spec-reminder-item.
- There is a failing test set up in tests/acceptance/reminder-list-test.js.

Hot Tip: If you put “Closes #1” (assuming the issue is marked #1) in the pull request, your commit, or anywhere in the body of your PR, it will automatically close this issue when it successfully merged into master.

### LINKED ISSUE #2: Reroute to Reminders
When the user visits the root of the application, they should be redirected to `'/reminders'` and should see a list of all reminders on the page. This means making a hopefully small adjustment to your `router.js` file.  

**Hot Tip**: This will make your previous test fail (if you had it passing). Make the necessary changes in the test(s).

### LINKED ISSUE #3: See Individual Reminder
When the user clicks on one of the reminders on the page, they are taken to an individual reminders route (ie: '/reminders/1', if the id of the reminder is `1`). The title of the reminder object should be displayed.

So, what needs to happen here?

* You'll need a `reminder` route in addition to the `reminders` route.
* It should be nested inside the `reminders` route.
* The user should still see all of the reminders from your previous feature. The specific reminder they are looking at now should be rendered in the outlet of the `reminders.hbs` template.
* There should be some CSS for the `.active` class.

There is a failing test set up in `tests/acceptance/reminder-list-test.js` to get you started - this test needs to be rewritten to reflect the route you are expecting to visit.

### LINKED ISSUE #5: Create A New Reminder
At the bottom of the `reminders.hbs` template, there should be a button that routes the user to `'/new'` where users can create a new reminder.

* The new reminder should have inputs for the title, date, and notes.
* The new reminder should have a button for submitting the new note.
* Pressing the button, should submit reminder to the database and have it shown in the Reminders list.

**Hot Tip**: TDD is really helpful.

### PR DESCRIPTION
**Pull request by Pilewski**
## Purpose

- At the bottom of the reminders.hbs template, there should be a button that routes the user to '/new' where users can create a new reminder.

- The new reminder should have inputs for the title, date, and notes.
- The new reminder should have a button for submitting the new note.
- Pressing the button, should submit reminder to the database and have it shown in the Reminders list.

## Approach

- Created an add-reminder component with a createReminder() action that will store a new reminder with the salient properties 
- Created a sub-route of new that links to a 'new' template that has the add-reminder component as its output  

### Learning

-Trolled around the ember docs and referred to Steve's finished ember-grocery repo

#### Before

Closes #1 
Closes #2 
Closes #3
Closes #5 

#### After

### CODE REVIEW

**Comment by Pilewski** [comment]
Thanks y'all 💪

File: app/components/add-reminder.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  store: Ember.inject.service(),
+
+  tagName: 'section',
```
Comment (nfosterky) [inline app/components/add-reminder.js]: 👍  for using tagName

File: app/router.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ const Router = Ember.Router.extend({
 Router.map(function() {
   this.route('reminders', function(){
     this.route('reminder', { path: ':reminder_id' });
+    this.route('new', { path: '/new' })
```
Comment (nfosterky) [inline app/router.js]: `{ path: 'new'}` is redundant here since the route is already `'new'`

File: app/templates/components/add-reminder.hbs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  </label>
+  <label>
+    Date
+    {{input type="date" class="spec-input-date" value=date}}
+  </label>
+  <label>
+    Notes
+    {{textarea class="spec-textarea-notes" value=notes}}
+  </label>
+  <button class="add-reminder--submit">Submit</button>
```
Comment (nfosterky) [inline app/templates/components/add-reminder.hbs]: fix your className

File: app/templates/reminders.hbs [hunk 4]
```diff
@@ -8,6 +8,13 @@
...
   {{/each}}
+  {{#link-to 'reminders.new'}}
+    <button> Add Reminder </button>
+  {{/link-to}}
 </ul>
 
+
+<!-- {{add-reminder}} -->
+
+
```
Comment (nfosterky) [inline app/templates/reminders.hbs]: Remove this extra whitespace

File: app/templates/reminders/new.hbs [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{{add-reminder}}
+
+{{outlet}}
```
Comment (nfosterky) [inline app/templates/reminders/new.hbs]: You can remove `{{outlet}}` if you are not using it

**Comment by nfosterky** [comment]
What is the status on this?

**Comment by ianlancaster** [comment]
Updates made @nfosterky! 👍

**Pull request merged**